hwloc/topology-rsmi.c

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright © 2012-2020 Inria.  All rights reserved.
+ * Copyright (c) 2020, Advanced Micro Devices, Inc. All rights reserved.
+ * Written by Advanced Micro Devices,
+ * See COPYING in top-level directory.
+ */
+
+#include "private/autogen/config.h"
+#include "hwloc.h"
+#include "hwloc/plugins.h"
+
+/* private headers allowed for convenience because this plugin is built within hwloc */
+#include "private/misc.h"
+#include "private/debug.h"
+
+#include <dlfcn.h>
+#include <rocm_smi.h>
+
+/**
+ * PCI information about a GPU device.
+ */
+typedef struct rsmiPciInfo_st {
+  union {
+    struct {
+#ifdef HWLOC_BIGENDIAN
+      uint64_t domain : 32;
+      uint64_t reserved : 16;
+      uint64_t bus : 8;
+      uint64_t device : 5;
+      uint64_t function : 3;
+#else
+      uint64_t function : 3;
+      uint64_t device : 5;
+      uint64_t bus : 8;
+      uint64_t reserved : 16;
+      uint64_t domain : 32;
+#endif
+    };
+    uint64_t bdfid;
+  };
+} rsmiPciInfo_t;
+
+/*
+ * Get the name of the GPU
+ *
+ * dv_ind		(IN) The device index
+ * device_name	(OUT) Name of GPU devices
+ * size			(OUT) Size of the name
+ */
+static void get_device_name(uint32_t dv_ind, char *device_name, unsigned int size)
+{
+  const char *status_string;
+  rsmi_status_t rsmi_rc = rsmi_dev_name_get(dv_ind, device_name, size);
+
+  if (rsmi_rc != RSMI_STATUS_SUCCESS) {
+    rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
+    fprintf(stderr, "Failed to get name of the GPU: %s\n", status_string);
```

Comment:
Make sure these error messages only occur under catastrophic circumstance sicne we don't want the hwloc library to pollute the stderr of programs that don't even know there are using hwloc.
Also add a ref to RSMI and/or rsmi_dev_name() in the message.
Same for all other fprintf(stderr...) below.

Reply:
OK